Sensu Go Deployment Checklist

Blue0ctober opened the pull request:
<!--- Thanks for submitting a pull request! -->
<!--- If you haven't yet, please read the contributing guide at: -->
<!--- https://github.com/sensu/sensu-docs/blob/master/CONTRIBUTING.md -->
<!--- Provide a general summary of your changes in the Title above. -->

## Description
A guide on hardware and architect recommendations for a robust Sensu Go deployment.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here: "Closes #555" -->
Closes #1074 

## Review Instructions
<!--- Optional -->

Blue0ctober committed:
inital push

apaskulin committed:
Merge branch 'master' into SGO-deployment-guide

apaskulin committed:
Merge branch 'master' into SGO-deployment-guide

Commit:
Merge branch 'master' into SGO-deployment-guide

apaskulin committed:
Merge branch 'master' into SGO-deployment-guide

apaskulin committed:
Merge branch 'master' into SGO-deployment-guide

apaskulin committed:
Merge branch 'master' into SGO-deployment-guide

Blue0ctober committed:
First pass rework

Blue0ctober committed:
Merge branch 'master' into SGO-deployment-guide

Blue0ctober committed:
Update deployment-guide.md

apaskulin commented on content/sensu-go/5.0/guides/deployment-guide.md:
```diff
@@ -0,0 +1,119 @@
+
+Load balancers are a great method to distribute traffic coming from sensu-agents to sensu-backends. Load balancers should be configured to evenly distribute traffic between all sensu-backends. Sticky sessions are useful but not required.
+
+TODO: VERIFY Utilization of healthz endpoint to check availablity of sensu-backends
```
https://docs.sensu.io/sensu-go/5.11/api/health/

apaskulin commented on content/sensu-go/5.0/guides/deployment-guide.md:
```diff
@@ -0,0 +1,119 @@
+_WARNING: Configuring SSL encryption post deployment will require reseting all etcd clusters which will result in datalost (eg. event data)_
+
+[1]: ../../installation/recommended-hardware
+[2]: TODO: RELAY HANDLER GITHUB LINK
```
https://bonsai.sensu.io/assets/sensu/sensu-relay-handler

Blue0ctober committed:
added diagrams and links

apaskulin committed:
Merge branch 'master' into SGO-deployment-guide

Commit:
Merge branch 'master' into SGO-deployment-guide

Commit:
spelling cleanup

Commit:
revising deployment guide

apaskulin committed:
review edits

- Use sentence case for titles
- linkTitles should be verb-first
- Moves introduction section to the beginning of the guide
- Typo fixes
- Add diagram captions and source comments

apaskulin committed:
add ports to diagram

apaskulin committed:
change title

apaskulin committed:
update diagram

apaskulin committed:
copy to all versions

apaskulin reviewed (approved):
Edited and copied to all versions!

apaskulin committed:
Merge branch 'master' into SGO-deployment-guide
